x-pack/plugins/data_enhanced/server/search/session/session_service.ts

```diff
@@ -111,41 +122,63 @@ export class BackgroundSessionService implements ISessionService {
     return session;
   };
 
-  // TODO: Throw an error if this session doesn't belong to this user
-  public get = (sessionId: string, { savedObjectsClient }: BackgroundSessionDependencies) => {
-    return savedObjectsClient.get<BackgroundSessionSavedObjectAttributes>(
+  public get = async (
+    user: AuthenticatedUser | null,
+    sessionId: string,
+    { savedObjectsClient }: BackgroundSessionDependencies
+  ) => {
+    const session = await savedObjectsClient.get<BackgroundSessionSavedObjectAttributes>(
       BACKGROUND_SESSION_TYPE,
       sessionId
     );
+    this.throwOnUserConflict(user, session);
+    return session;
   };
 
-  // TODO: Throw an error if this session doesn't belong to this user
-  public find = (
+  public find = async (
+    user: AuthenticatedUser | null,
     options: BackgroundSessionFindOptions,
     { savedObjectsClient }: BackgroundSessionDependencies
   ) => {
+    const userFilters =
```

Comment:
As we're scoped to a session, aren't saved objects restricted by user automatically?
And if not, shouldn't they be? :thinking:

Reply:
+1 to this question